### ISSUE TITLE
Popover's boundary prop does not accept Element objects

### ISSUE THREAD

**Issue by mmermerkaya** [palantir/blueprint#3570]
**Package:** `@blueprintjs/core@3.15.1`

**Issue:** The `boundary` field of `IPopoverSharedProps` ([defined here](https://github.com/palantir/blueprint/blob/4908bb068317ba1eb7e338034bc1195be7251a6e/packages/core/src/components/popover/popoverSharedProps.ts#L42)) does not accept values of type `Element`.

**Description:** This field (`boundary`) is only used as the `boundariesElement` field in `flip` and `preventOverflow` objects in Popper modifiers, as seen [here](https://github.com/palantir/blueprint/blob/4908bb068317ba1eb7e338034bc1195be7251a6e/packages/core/src/components/popover/popover.tsx#L397). Both of these fields in Popper (defined [here](https://github.com/FezVrasta/popper.js/blob/36cc6c66e55f716232166c9a5725d8d08c4ec338/packages/popper/index.d.ts#L64) and [here](https://github.com/FezVrasta/popper.js/blob/36cc6c66e55f716232166c9a5725d8d08c4ec338/packages/popper/index.d.ts#L74)) accept objects of type `Boundary | Element`, but the `boundary` field defined in Blueprint only accepts `Boundary`.

**Possible Solution:** Change the type [here](https://github.com/palantir/blueprint/blob/4908bb068317ba1eb7e338034bc1195be7251a6e/packages/core/src/components/popover/popoverSharedProps.ts#L42) from `PopperBoundary` to `PopperBoundary | Element`.

**Comment by adidahiya** [comment]
Good call, I think your solution would work. Would you be open to sending a quick PR? Ideally, add a small unit test to make sure it doesn't break the Popper instance at runtime.
